### PR TITLE
Followup from PR review on #386

### PR DESCRIPTION
**Pull request by dcoutts**
@jorisdral had some excellent feedback on PR #386 after it got merged. This PR addresses the review feedback https://github.com/IntersectMBO/lsm-tree/pull/386#pullrequestreview-2321588795. Roughly one patch per comment.

### CODE REVIEW

**Comment by dcoutts** [comment]
Oh, one patch I was meaning to add here was one to specialise all the WBB functions to IO.

File: src/Database/LSMTree/Internal/WriteBufferBlobs.hs [hunk 1]
```diff
@@ -211,53 +180,40 @@ readBlob :: (PrimMonad m, MonadThrow m)
...
+    return (SerialisedBlob rb)
 
-      -- We must not be asking to read blobs if no blobs have been written.
-      -- That would be an internal error.
-      NoBlobFileYet{} ->
-        error "BlobFile.mkBlobRef: no such blob refs, no blob file open"
 
 -- | Helper function to make a 'BlobRef' that points into a 'WriteBufferBlobs'.
 -- TODO: this two stage function design isn't terribly nice, and allocates
 -- more than necessary (a closure every call).
```
Comment (jorisdral) [inline src/Database/LSMTree/Internal/WriteBufferBlobs.hs:198]:
```suggestion
```

File: doc/format-directory.md [hunk 2]
```diff
@@ -42,12 +42,29 @@ in use by any open LSM handle in the session.
...
 snapshots). This is because the metadata is held in memory for open LSM handles.
 
+# Write buffer blobs files
+
+The `${n}.wbblobs` file format is the same as the `${n}.blobs` format used for
+runs: the concatenation of all the blobs written out so far.
+
+One difference is that these files are shared between open tables created using
+the `duplicate` operation, and so can contain the blobs inserted via many table
+handles.
```
Comment (jorisdral) [inline doc/format-directory.md:63]: Technically, blob files for runs are also shared on `duplicate`, but it's different in the sense that many tables might write to the same write buffer blob file, whereas the run blob files are immutable

**Review by jorisdral — APPROVED** [review]
LGTM!

File: src/Database/LSMTree/Internal/WriteBufferBlobs.hs (hunk 1 above)
Comment (dcoutts) [reply src/Database/LSMTree/Internal/WriteBufferBlobs.hs:198]: Yes indeed.

File: doc/format-directory.md (hunk 2 above)
Comment (dcoutts) [reply doc/format-directory.md]: I've revised the wording.

**Pull request merged**